cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs

```diff
@@ -1706,16 +1706,21 @@ impl_runtime_apis! {
 				Config as XcmBridgeHubRouterConfig,
 			};
 
+			pub const RANDOM_ID: u32 = 43211234;
+			pub const fn CREATE_RND_LOC() -> Location {
+					Location::new(
+						1,
+						[Parachain(ParaId::new(RANDOM_ID).into())]
+					)
+			};
 			parameter_types! {
+				pub const RandomParaId: u32 = RANDOM_ID;
+				pub const RandomParaLocation: Location = CREATE_RND_LOC();
```

Comment:
Please keep it simple and consistent across all runtimes, we don't need/want here another `const` or `const fn`, just this is enough:
```			
				pub const RandomParaId: ParaId = ParaId::new(43211234);
				pub const RandomParaLocation: Location = ParentThen(Parachain(RandomParaId::get().into()).into()).into();
```

Reply:
Hey @bkontur 
I was testing this because I'm encountering an E00015 error—running a non-const in a const context. For some reason, I’m unable to reproduce some of these issues locally even when I run cargo check | cargo check --lib | cargo clippy | clearing my cache.

Currently, I'm examining both RandomParaId and RandomParaLocation to ensure that all abstracted code remains within a single context without unintended mixing.

Reply:
what about removing just `const` from  `pub const RandomParaLocation`?

```suggestion
				pub const RandomParaId: ParaId = ParaId::new(43211234);
				pub RandomParaLocation: Location = ParentThen(Parachain(RandomParaId::get().into()).into()).into();
```

Reply:
Almost, this approach would work but at AST (Abstract Syntax Tree) level, Rust would detect a mismatch in const context when RandomParaId is called in the RandomParaLocation ... ParaId::new(43211234) is not not const compatible to.

My next push is 
```
pub RandomParaId: ParaId = ParaId::new(43211234);

pub RandomParaLocation: Location = ParentThen(Parachain(RandomParaId::get().into()).into()).into();
```

Also, @bkontur  also another area is the `TrustedTeleporter`. For potental E00015 because  `TrustedTeleporter` has the const keyword added to it.
Gonna push that and see.

Reply:
If the current implementation faces compilation issues, I suggest creating an issue to improve the const-compatibility of ParentThen and its .into() methods. This update would maintain backward compatibility with existing runtime implementations while extending const evaluation capabilities.

Reply:
![Screenshot 2025-02-06 at 19 52 55](https://github.com/user-attachments/assets/223f66f6-5d24-4132-9748-f37f63d51247)

Pushing a fix though

Reply:
> Also, @bkontur also another area is the `TrustedTeleporter`. For potental E00015 because `TrustedTeleporter` has the const keyword added to it. 

please, just remove `const` from `pub const TrustedTeleporter` or wherever, and just use simple:
```
pub const RandomParaId: ParaId = ParaId::new(43211234);
pub RandomParaLocation: Location = ParentThen(Parachain(RandomParaId::get().into()).into()).into();

...

pub TrustedTeleporter: Option<(Location, Asset)> = Some((
        RandomParaLocation::get(),
        Asset { fun: Fungible(UNITS), id: AssetId(TokenLocation::get())
));
```

Reply:
Understood